controllers/managedocs_controller.go

```diff
@@ -44,71 +45,131 @@ import (
 )
 
 const (
-	storageClusterName  = "ocs-storagecluster"
+	storageClusterName     = "ocs-storagecluster"
+	prometheusName         = "managed-ocs-prometheus"
+	alertmanagerName       = "managed-ocs-alertmanager"
+	alertmanagerConfigName = "managed-ocs-alertmanager-config"
+
 	storageClassSizeKey = "size"
 	deviceSetName       = "default"
+	monLabelKey         = "app"
+	monLabelValue       = "managed-ocs"
 )
 
 // ManagedOCSReconciler reconciles a ManagedOCS object
 type ManagedOCSReconciler struct {
 	client.Client
+
 	Log                  logr.Logger
 	Scheme               *runtime.Scheme
 	AddonParamSecretName string
 
-	ctx        context.Context
-	managedOCS *v1.ManagedOCS
-	namespace  string
+	ctx            context.Context
+	managedOCS     *v1.ManagedOCS
+	storageCluster *ocsv1.StorageCluster
+	prometheus     *promv1.Prometheus
+	alertmanager   *promv1.Alertmanager
```

Comment:
This is not the actual case, as we need all of them also in `getComponentStatus` and later on in the deletion code. 
The reconciler object is acting as a context object for the entire operation and should hold to everything that is not immediately local

Reply:
True, I didn't think about the deletion code. I think it would be better for clarity to move the context objects into its own structure within the ManagedOCSReconciler object, but I'm fine if you disagree.

Reply:
So what will stay inside the reconciler, as I can argue that everything there is context. include the logger and the client.
Saying that I am resolving it as is